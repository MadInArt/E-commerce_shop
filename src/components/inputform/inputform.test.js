  
import React from 'react';
import { shallow } from 'enzyme';

import InputForm from './inputform';

describe('InputForm component', () => {
  let wrapper;
  let mockHandleChange;

  beforeEach(() => {
    mockHandleChange = jest.fn();

    const mockProps = {
      label: 'email',
      value: 'test1@gmail.com',
      onChangeHandle: mockHandleChange
    };

    wrapper = shallow(<InputForm {...mockProps} />);
  });

  test('should render InputForm component', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('should call handleChange method when input changes', () => {
    wrapper.find('FormInputContainer').simulate('change');

    expect(mockHandleChange).toHaveBeenCalled();
  });

  test('should render FormInputLabel if there is a label', () => {
    expect(wrapper.exists('FormInputLabel')).toBe(true);
  });

  test('should not render FormInputLabel if there is no label', () => {
    const mockNewProps = {
      label: '',
      value: 'test1@gmail.com',
      onChangeHandle: mockHandleChange
    };

    const newWrapper = shallow(<InputForm {...mockNewProps} />);

    expect(newWrapper.exists('FormInputLabel')).toBe(false);
  });
});