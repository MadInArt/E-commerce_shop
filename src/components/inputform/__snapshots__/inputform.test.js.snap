// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputForm component should render InputForm component 1`] = `
<styled.div>
  <FormInputContainer
    onChange={[MockFunction]}
    value="test1@gmail.com"
  />
  <FormInputLabel
    className="shrink"
  >
    email
  </FormInputLabel>
</styled.div>
`;
